### PR TITLE
1265 parallel api

### PR DESCRIPTION
**Pull request by mvandeberg**

### CODE REVIEW

**Comment by theoreticalbts** [comment]
- "Remove unnecessary indices #1078" should be separate PR.
- Spot refactors of database_api in 69403684cc3da5b4248f0e4fe47b0fe712973d96 should be separate PR.
- Commits creating the `db_api` plugin should be their own PR.

**Comment by mvandeberg** [comment]
Realistically, refactoring those commits into their own PRs and then rebasing this branch back on master is going to be significant work to get us where we are right now, but with a prettier git history. While I value having a pretty git history, it is not worth the minimum 1 day of work (possibly more) to get things there.

**Comment by theoreticalbts** [comment]
I just finished going over that entire diff.

CI is succeeding, but failing to communicate with Github.  Error message:  "Could not update commit status, please check if your scan credentials belong to a member of the organization or a collaborator of the repository and repo:status scope is selected".

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "libraries/fc"]
```
Comment (sneak) [inline .gitmodules]: should we vendor these in instead of using submodules?  especially considering that we are now pulling in an external one?

**Comment by mvandeberg** [comment]
-[ ] Remove cpp-netlib submodule
-[ ] Check appbase cpp14 requirement. If required, update documentation. Defer?
-[ ] Move comment content patch into branch

**Comment by mvandeberg** [comment]
`witness_plugin_impl` is now in the detail namespace. It fixes large parts of the diff, but not all because the witness plugin in master, has a mix of methods in the plugin class and impl. This version of the plugin has everything in impl. It would probably be easiest to review with a manual side by side inspection of the relevant methods.

**Comment by theoreticalbts** [comment]
Approved.

**Pull request merged**